core/templates/dev/head/domain/exploration/ReadOnlyExplorationBackendApiService.js

```diff
@@ -14,25 +14,35 @@
 
 /**
  * @fileoverview Service to retrieve read only information
- * about exploraions from the backend.
+ * about explorations from the backend.
  */
 
 oppia.factory('ReadOnlyExplorationBackendApiService', [
   '$http', '$q', 'EXPLORATION_DATA_URL_TEMPLATE',
-  'UrlInterpolationService',
+  'EXPLORATION_VERSION_DATA_URL_TEMPLATE', 'UrlInterpolationService',
   function($http, $q, EXPLORATION_DATA_URL_TEMPLATE,
-    UrlInterpolationService) {
+    EXPLORATION_VERSION_DATA_URL_TEMPLATE, UrlInterpolationService) {
     // Maps previously loaded explorations to their IDs.
     var _explorationCache = [];
 
     var _fetchExploration = function(
-        explorationId, successCallback, errorCallback) {
-      var explorationDataUrl = UrlInterpolationService.interpolateUrl(
+        explorationId, version, successCallback, errorCallback) {
+      var explorationDataUrl = '';
+      if (version) {
+        explorationDataUrl = UrlInterpolationService.interpolateUrl(
+        EXPLORATION_VERSION_DATA_URL_TEMPLATE, {
+          exploration_id: String(explorationId),
```

Comment:
Currently it's only necessary due to tests in HistoryServiceSpec.js, as mentioned above.